sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/csv/CSVInferSchema.scala

```diff
@@ -178,7 +164,8 @@ class CSVInferSchema(val options: CSVOptions) extends Serializable {
     // We can only parse the value as TimestampNTZType if it does not have zone-offset or
     // time-zone component and can be parsed with the timestamp formatter.
     // Otherwise, it is likely to be a timestamp with timezone.
-    if ((allCatch opt timestampNTZFormatter.parseWithoutTimeZone(field, false)).isDefined) {
+    val fieldAsUTF8String = UTF8String.fromString(field)
+    if (DateTimeUtils.stringToTimestampWithoutTimeZone(fieldAsUTF8String).isDefined) {
```

Comment:
hmm, shouldn't we respect the datetime format string that the user specified? otherwise we may not be able to read the datetime value even if we can infer it.

Reply:
Yes, that's why tests failed

Reply:
Yes, I left a similar comment above, we would like to retain the ability to use options to configure timestamp format. One way to do it could be adding a couple of more methods to TimestampFormatter trait to allow returning an option where `None` means could not parse and `Some(value)` means parsing succeeded.